tests/basic/telemetry_test.go

```diff
@@ -271,6 +273,8 @@ var _ = Describe("{ProfileOnlyDiags}", func() {
 						return
 					}
 				})
+				// This sleep is required because the heap and stack logs might not have been written at same time.
+				time.Sleep(30* time.Second)
```

Comment:
I would do retry here instead of hardcoded sleep

Reply:
The sleep can actually be removed.  The next step does an "Eventually" which retries the check for the newly generated stack/heap files so no need to have this sleep here at all.